corehq/apps/app_manager/views/apps.py

```diff
@@ -196,15 +203,33 @@ def get_app_view_context(request, app):
             app_ver = MAJOR_RELEASE_TO_VERSION[option.build.major_release()]
             builds["default"] = build_config.get_default(app_ver).to_string()
 
-    (build_spec_setting,) = filter(
-        lambda x: x['type'] == 'hq' and x['id'] == 'build_spec',
-        [setting for section in settings_layout
-            for setting in section['settings']]
-    ) if settings_layout else (None,)
+    def _get_setting(setting_type, setting_id):
+        # get setting dict from settings_layout
+        if not settings_layout:
+            return None
+        matched = filter(
+            lambda x: x['type'] == setting_type and x['id'] == setting_id,
+            [
+                setting for section in settings_layout
+                for setting in section['settings']
+            ]
+        )
+        if matched:
+            return matched[0]
+        else:
+            return None
+
+    build_spec_setting = _get_setting('hq', 'build_spec')
     if build_spec_setting:
         build_spec_setting['options_map'] = options_map
         build_spec_setting['default_app_version'] = app.application_version
 
+    practice_user_setting = _get_setting('hq', 'practice_mobile_worker_id')
+    if has_privilege(request, privileges.PRACTICE_MOBILE_WORKERS) and practice_user_setting:
```

Comment:
`practice_user_setting ` will be `None` if the min app version is not met, so it's indirectly checked. I have swapped the two conditions to avoid privilege check when not necessary